pkg/playback/http_playback.go

```diff
@@ -0,0 +1,660 @@
+package playback
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+func handleErrorInHandler(w http.ResponseWriter, err error, statusCode int) {
+	if err == nil {
+		return
+	}
+
+	// This line will log a relatively benign 'superfluous response.WriteHeader' line if
+	// w.WriteHeader was already called on this http.ResponseWriter. Ideally, response.WriteHeader
+	// should not be called before this function is called, since the HTTP response code can only be
+	// set once.
+	// TODO: refactor usage of this function so that we do not get this 'superfluous' log message
```

Comment:
Let's do this now

Reply:
Addressed this in 0bb511c5b8c910a9d01be9002f79e3364150897c.